src/constants/env.ts

```diff
@@ -263,876 +267,80 @@ const Config = ss.object({
-      }),
-    });
-  }
-  if (env.NEXT_PUBLIC_SUBGRAPH_V2_1116) {
-    subgraphs.push({
-      source: "gql",
-      url: env.NEXT_PUBLIC_SUBGRAPH_V2_1116,
-      type: "Optimistic Oracle V2",
-      chainId: 1116,
-      address: getContractAddress({
-        chainId: 1116,
-        type: "Optimistic Oracle V2",
-      }),
-    });
-  }
-  if (env.NEXT_PUBLIC_SUBGRAPH_V2_42161) {
-    subgraphs.push({
-      source: "gql",
-      url: env.NEXT_PUBLIC_SUBGRAPH_V2_42161,
-      type: "Optimistic Oracle V2",
-      chainId: 42161,
-      address: getContractAddress({
-        chainId: 42161,
-        type: "Optimistic Oracle V2",
-      }),
-    });
-  }
-  if (env.NEXT_PUBLIC_SUBGRAPH_V2_5) {
-    subgraphs.push({
-      source: "gql",
-      url: env.NEXT_PUBLIC_SUBGRAPH_V2_5,
-      type: "Optimistic Oracle V2",
-      chainId: 5,
-      address: getContractAddress({ chainId: 5, type: "Optimistic Oracle V2" }),
-    });
-  }
-  if (env.NEXT_PUBLIC_SUBGRAPH_V2_80001) {
-    subgraphs.push({
-      source: "gql",
-      url: env.NEXT_PUBLIC_SUBGRAPH_V2_80001,
-      type: "Optimistic Oracle V2",
-      chainId: 80001,
-      address: getContractAddress({
-        chainId: 80001,
-        type: "Optimistic Oracle V2",
-      }),
-    });
-  }
-  if (env.NEXT_PUBLIC_SUBGRAPH_V2_11155111) {
-    subgraphs.push({
-      source: "gql",
-      url: env.NEXT_PUBLIC_SUBGRAPH_V2_11155111,
-      type: "Optimistic Oracle V2",
-      chainId: 11155111,
-      address: getContractAddress({
-        chainId: 11155111,
-        type: "Optimistic Oracle V2",
-      }),
-    });
+  for (const [key, value] of Object.entries(env)) {
+    if (!value) continue;
+    const [item, version, chainId] = key.split("_").slice(-3);
+    if (item === "SUBGRAPH") {
+      if (version === "SKINNY") {
+        const subgraph = {
+          source: "gql",
+          url: value,
+          type: "Skinny Optimistic Oracle",
+          chainId: parseInt(chainId),
+          address: getContractAddress({
+            chainId: parseInt(chainId),
+            type: "Skinny Optimistic Oracle",
+          }),
+        };
+        if (ss.is(subgraph, SubgraphConfig)) {
+          subgraphs.push(subgraph);
+        }
+      } else {
+        const subgraph = {
+          source: "gql",
+          url: value,
+          type: `Optimistic Oracle ${version}`,
+          chainId: parseInt(chainId),
+          address: getContractAddress({
+            chainId: parseInt(chainId),
+            type: `Optimistic Oracle ${version}`,
+          }),
+        };
+        if (ss.is(subgraph, SubgraphConfig)) {
+          subgraphs.push(subgraph);
+        }
+      }
+    }
+    if (item === "PROVIDER") {
+      if (version === "SKINNY") {
+        const provider = {
+          source: "provider",
+          type: "Skinny Optimistic Oracle",
+          url: value,
+          chainId: parseInt(chainId),
+          address: getContractAddress({
+            chainId: parseInt(chainId),
```

Comment:
@daywiss I need your sign off on this. Adding config items for Sepolia proved extremely tedious and will be the case for any chain we add. So if we follow the structure outlined in the readme for these variables then this function will be able to parse the env file and find the correct config items.

Reply:
the structure must be `NEXT_PUBLIC_("SUBGRAPH" | "PROVIDER")_(VERSION | "SKINNY")_(CHAINID)`